### DISCUSSION TITLE
Utilizing timestamping to synchronize mutiple ADALM plutos

### DISCUSSION THREAD

**Discussion by Argonghost** [srsran/zynq_timestamping#65] [General]
Hi, came across this page and i believe this is going to help me with my task, i just donno how to go about it given im a newbie.
So i would like to phase align multiple ADALM pluto sdr's, not an RF level but just at an FPGA level.  According to this discussion on Analog Devices: https://ez.analog.com/adieducation/university-program/f/q-a/164715/pluto-sdr-phase-reference-signal-for-lo, this seems to be the way to go.  Im just not sure how to proceed exactly after building image into both pluto SDR's.  How many PRB is needed for correct time stamping such that i can ensure timing alignment between two zynq's ??

The idea is that once timing alignment between two zynqs is ensured, DSP techniques can be done at baseband to ensure phase offset correction.  

SIncerely,
AG

**Comment by ofontbach** [comment]
Hi @Argonghost,

The timestamping mechanism is thought to create local time references by counting the clock ticks at the FPGA. Hence, if you deploy it in two different pluto SDRs, each will have an independent time reference, which you won't be able to correlate straightforwardly. In other words, the timestamping mechanism was not designed to phase-align multiple SDR boards.

For time alignment, you would need to extend the design towards that end, so that the timestamping information between the two could be exchanged and the required offset applied on one of the two. Additionally, for phase alignment, both pluto SDRs would need to share an external reference clock.

Cheers

**Reply by stan3c** [reply]
Hi, I am also interested in the possibility to synchronize multiple Pluto boards. What I am looking for is to fully synchronize the boards for use in beam forming application. 
I have Pluto  derivatives with 10MHz and 1PPS inputs and I am looking for a way to synchronize them with time stamping the same way it is done in UHD driver form Ettus.
Is this already possible with zynq_timestamping? I understand it would work in a different way than UHD but as long as the effect is usable for beam forming I can accept it.

**Comment by stan3c** [comment]
Hi again,  I read that the guys that make AntSDR have ported UHD driver to AntSDR. There is a version of firmware that supports UHD timestamping compiled for AntSDR. 
Am I wrong that it should be fairly easy to port the AntSDR UHD version of firmware to Pluto SDR? Or maybe it has been done already?
